cli/src/runner.rs

```diff
@@ -0,0 +1,179 @@
+use std::io;
+use std::ops::Deref;
+use std::path::PathBuf;
+
+use analyzer::diagnostic::Diagnostic;
+use analyzer::engine::Engine;
+use analyzer::importer::ASTImporter;
+use analyzer::name::Name;
+use analyzer::relations::{Definition, SourceId};
+use analyzer::steps::typing::apply_types;
+use analyzer::types::engine::CodeEntry;
+use ast::group::Block;
+use ast::Expr;
+use compiler::compile;
+use context::source::{Source, SourceSegmentHolder};
+use lexer::lexer::lex;
+use parser::err::ParseReport;
+use parser::parse;
+
+use crate::cli::{display_exprs, display_tokens, execute, Configuration};
+use crate::display_bytecode::display_bytecode;
+use crate::formatted_diagnostic::render_diagnostic;
+use crate::formatted_parse_error::render_parse_error;
+use crate::source_importer::FileSourceImporter;
+
+pub(crate) fn run(source: PathBuf, working_dir: PathBuf, config: Configuration) -> bool {
+    let name = source.to_string_lossy();
+    let mut name = name.deref().replace('/', "::");
+    if let Some((name_no_extension, _)) = name.rsplit_once('.') {
+        name = name_no_extension.to_string()
+    }
+    let entry_point = Name::new(&name);
+
+    let mut importer = RunnerImporter::new(working_dir);
+
+    let result = analyzer::resolve_all(entry_point, &mut importer);
+
+    let mut diagnostics = result.diagnostics;
+    let engine = result.engine;
+    let relations = result.relations;
+
+    if !importer.errors.is_empty() {
+        display_import_errors(importer.errors);
+        return true;
+    }
+
+    let typed_engine = apply_types(&engine, &relations, &mut diagnostics);
+
+    if !diagnostics.is_empty() {
+        display_diagnostics(diagnostics, &engine, &importer);
+        return true;
+    }
+
+    let mut bytecode = Vec::new();
+    let root_expr = typed_engine
+        .get(Definition::User(SourceId(0)))
+        .map(|c| match c {
+            CodeEntry::User(c) => &c.expression,
+            _ => unreachable!(),
+        })
+        .unwrap();
```

Comment:
```suggestion
    let root_expr = &typed_engine.get_user(SourceId(0)).unwrap().expression;
```
This is also duplicated with the REPL runner.